src/WorkflowCore/Services/ExecutionResultProcessor.cs

```diff
@@ -110,7 +110,7 @@ private void Compensate(WorkflowInstance workflow, WorkflowDefinition def, Execu
             exceptionPointer.EndTime = _datetimeProvider.Now.ToUniversalTime();
             exceptionPointer.Status = PointerStatus.Failed;
 
-            while (scope.Any())
+            while (scope.Count > 0)
```

Comment:
I did that  because of performance since `Count` has a backing filed and it's value is already there, but `Any()` uses `Enumerator`. So Should I change this back?

Reply:
Hi, if its any help here is the implementation of both:
- [Stack<>.Count using a backing field](https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/Generic/Stack.cs#L63)
- [Enumerable<>.Any() using an enumerator](https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/AnyAll.cs#L20)